setup_deb.sh

```diff
@@ -68,133 +68,142 @@ log "Installing python dependencies ..."
 sleep 1
 # direct dependencies
 pip3 install --no-deps --upgrade \
-   SimpleITK==2.2.1 \
-   batchgenerators==0.21 \
-   medpy==0.4.0 \
-   nibabel==3.2.2 \
-   "numpy<1.24.0" \
-   opencv-python==4.8.0.76 \
-   pandas==1.4.2 \
-   pycocotools==2.0.6 \
-   scikit-build==0.14.1 \
-   scipy==1.8.0 \
-   tifffile==2023.1.23.1 \
-   tqdm \
-   sacrebleu==2.3.1 \
-   sentencepiece==0.1.97 \
-   tiktoken==0.3.3 \
-   ultralytics==8.0.75 \
-   evaluate==0.4.0 \
-   datasets==2.13.1 \
-   soundfile==0.12.1 \
-   librosa==0.10.0.post2 \
-   numba==0.59.0 \
-   py-cpuinfo==9.0.0 \
-   cchardet==2.1.7
+    SimpleITK==2.2.1 \
+    batchgenerators==0.21 \
+    medpy==0.4.0 \
+    nibabel==3.2.2 \
+    "numpy<1.24.0" \
+    opencv-python==4.8.0.76 \
+    pandas==1.4.2 \
+    pycocotools==2.0.6 \
+    scikit-build==0.14.1 \
+    scipy==1.8.0 \
+    tifffile==2023.1.23.1 \
+    tqdm \
+    sacrebleu==2.3.1 \
+    sentencepiece==0.1.97 \
+    tiktoken==0.3.3 \
+    ultralytics==8.0.75 \
+    evaluate==0.4.0 \
+    datasets==2.13.1 \
+    soundfile==0.12.1 \
+    librosa==0.10.0.post2 \
+    numba==0.59.0 \
+    py-cpuinfo==9.0.0 \
+    cchardet==2.1.7
+
+python3 -c '
+import sys
+version = sys.version_info
+if version.major == 3 and version.minor == 11:
+    print("pip3 install --no-deps --upgrade scipy==1.10.0")
+else:
+    print("pip3 install --no-deps --upgrade scipy==1.8.0")
+' | sh
```

Comment:
I'm not a fan of this. I think we should instead move all Python requirements to a requirements.txt file and use environment markers in cases like this https://peps.python.org/pep-0508/#environment-markers

Reply:
Yes, I agree. Using requirements.txt with 'python_version' marker will be much more elegant solution. I'm using it in manifests repo. However, it's up to you, If I can make this change, as it's big one.

Reply:
I think the behaviour should stay the same if we keep all the flags and versions the same, so it wouldn't be that big of a change. @jan-grzybek-ampere what do you think?